File: backend/src/main/java/com/bangchat/data/entity/SpeechStyle.java
package com.bangchat.data.entity;

import lombok.Data;

/**
 * 角色说话语气实体，映射 speech_style 表。
 * 每个角色最多一条记录，描述其基础说话风格和语气特征。
 */
@Data
public class SpeechStyle {
    private Long id;
    private String characterId;
    private String description;
}

File: bangchat/lib/models/speech_style.dart
/// 角色说话语气实体，映射 speech_style 表。
class SpeechStyle {
  final int? id;
  final String characterId;
  final String description;

  const SpeechStyle({
    this.id,
    required this.characterId,
    required this.description,
  });

  factory SpeechStyle.fromMap(Map<String, dynamic> map) {
    return SpeechStyle(
      id: map['id'] as int?,
      characterId: (map['character_id'] as String?) ?? '',
      description: (map['description'] as String?) ?? '',
    );
  }
}
